test/functional/wallet_balance.py

```diff
@@ -140,6 +152,48 @@ def run_test(self):
         after = self.nodes[1].getunconfirmedbalance()
         assert_equal(before + Decimal('0.1'), after)
 
+        # Create 3 more wallet txs, where the last is not accepted to the
+        # mempool because it is the third descendant of the tx above
+        for _ in range(3):
+            # Set amount high enough such that all coins are spent by each tx
+            txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 99)
+        assert txid not in self.nodes[0].getrawmempool()
+        assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted
+
+        self.log.info("Test replacement and reorg of non-mempool tx")
+        tx_orig = self.nodes[0].gettransaction(txid)['hex']
+        # Increase fee by 1 coin
+        tx_replace = tx_orig.replace(
+            struct.pack("<q", 99 * 10**8).hex(),
+            struct.pack("<q", 98 * 10**8).hex(),
+        )
+        tx_replace = self.nodes[0].signrawtransactionwithwallet(tx_replace)['hex']
+        # Total balance is given by the sum of outputs of the tx
+        total_amount = sum([o['value'] for o in self.nodes[0].decoderawtransaction(tx_replace)['vout']])
+        self.sync_all()
+        self.nodes[1].sendrawtransaction(hexstring=tx_replace, maxfeerate=0)
+
+        # Now confirm tx_replace
+        block_reorg = self.nodes[1].generatetoaddress(1, ADDRESS_WATCHONLY)[0]
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(minconf=0), total_amount)
+
+        # Put txs back in mempool
```

Comment:
Presumably the txs don't get back into node 0's mempool because of the descendant limit, which is why the getbalance values are 0 below? If that's true, can you update this comment to explain what's going on?

Reply:
Added logprint with comment